File: aquality-appium-java-master/src/test/java/screens/SelectCityScreen.java
package screens;

import aquality.appium.elements.interfaces.IButton;
import aquality.appium.elements.interfaces.ILabel;
import aquality.appium.elements.interfaces.ITextBox;
import aquality.appium.screens.AndroidScreen;
import enums.CityItem;
import org.openqa.selenium.By;

public class SelectCityScreen extends AndroidScreen {

    private final IButton btnOkOnAlert = getElementFactory().getButton(By.id("android:id/button1"), "Button Ok on alert label");
    private final ILabel lblFirstAvailableCity = getElementFactory().getLabel(By.id("tvCityItemName"), "Label first available city");
    private final ITextBox textBoxSearchCity = getElementFactory().getTextBox(By.id("etSearchTest"), "Text box for search city");

    public SelectCityScreen() {
        super(By.id("action_bar_root"), "Select city screen");
    }

    public void clickOkOnAlertLabel() {
        btnOkOnAlert.click();
    }

    public void inputNameCity(CityItem cityItem) {
        textBoxSearchCity.clearAndType(cityItem.getCityItem());
        selectFirstAvailableCity();
    }

    public void selectFirstAvailableCity() {
        lblFirstAvailableCity.click();
    }

    public String getNameFirstAvailableCity() {
        return lblFirstAvailableCity.getText();
    }
}

File: KotlinSecretCloset/src/test/kotlin/screens/SelectCityScreen.kt
package screens

import enums.CityItem
import framework.elements.Button
import framework.elements.Label
import framework.elements.TextBox
import framework.screens.BaseScreen
import org.openqa.selenium.By

class SelectCityScreen : BaseScreen(By.id("action_bar_root"), "Select city screen") {

    private val btnOkOnAlert: Button =
        Button(By.id("android:id/button1"), "Button Ok on alert label")
    private val lblFirstAvailableCity: Label =
        Label(By.id("tvCityItemName"), "Label first available city")
    private val textBoxSearchCity: TextBox =
        TextBox(By.id("etSearchTest"), "Text box for search city")

    fun clickOkOnAlertLabel() = btnOkOnAlert.click()

    fun inputNameCity(cityItem: CityItem?) {
        textBoxSearchCity.clearAndType(cityItem!!.city)
        selectFirstAvailableCity()
    }

    private fun selectFirstAvailableCity() = lblFirstAvailableCity.click()
}